### ISSUE TITLE
Support `no_std` logging with `defmt`

### ISSUE THREAD

**Issue by rockboynton** [mcmah309/error_set#5]
Would be nice to have this option: https://defmt.ferrous-systems.com/introduction
gated by a feature flag

**Comment by mcmah309** [comment]
Good idea! I made the necessary changes and added the feature flag `defmt` to master. It should work. Would you mind testing it before I release a new version?

A good place to test would be here https://github.com/mcmah309/error_set/tree/master/test_no_std I'm not familiar with no_std so if you have any suggestions for the test suite, feel free to make a PR!

**Comment by mcmah309** [comment]
Added in [v0.6.0](https://github.com/mcmah309/error_set/releases/tag/v0.6.0). Closing for now, feel free to create another Issue/PR if you have suggestions for the implementation or tests :+1:
